packages/app/src/cli/models/extensions/specifications/app_access.ts

```diff
@@ -0,0 +1,27 @@
+import {BaseSchema} from '../schemas.js'
+import {createExtensionSpecification} from '../specification.js'
+import {zod} from '@shopify/cli-kit/node/schema'
+
+const AppAccessSchema = BaseSchema.extend({
```

Comment:
I tried to re-use the AppSchema by doing
```
const AppAccessSchema = BaseSchema.merge(AppSchema.pick({access: true}))
```
But this cause a few unit tests to fail with the `TypeError: Cannot read properties of undefined (reading 'pick')`. This failing for tests that don't mock out the app model. For example, tests inside `src/cli/services/dev/extension/server/utilities.test.ts` would fail.

Reply:
Paired with Alvaro to fix this